frontend/user/vue-components/ApplicationView.vue

```diff
@@ -84,13 +83,9 @@
         return this.currentApp.appData.image.policy;
       },
       appSource: function() {
-        let url = urlUtils.pathJoin(
-          window.apidata.base_url,
-          'containers',
-          this.currentApp.appData.container.url_id
-        );
-        let output = this.currentApp.delayed ? url : url + '/';
+        let url = this.$options.filters.appUrl(this.currentApp);
```

Comment:
you are misusing filters here. filters are for templates, with the pipe notation. You never use that filter in a template context. It is a generic shared routine.